node-lib/src/runner.rs

```diff
@@ -333,6 +333,7 @@ fn regtest_chain_config(options: &ChainConfigOptions) -> Result<ChainConfig> {
         chain_max_block_size_with_standard_txs,
         chain_max_block_size_with_smart_contracts,
         chain_pos_netupgrades,
+        chain_genesis_staking_settings,
```

Comment:
We should add guards such that if someone tries to pass staking settings, net upgrades, emission schedule, coin decimals, target spacing, and any other feature that's made for testing, in anything other than regtest, then we crash the software. This is important to prevent scammers from trying to get victims to create fake nodes somehow with custom settings and make them feel they got rich or something.

Reply:
It looks like `node-lib/src/regtest_options.rs`, where all of these "dangerous" parameters are defined to be read from config, are only used in RegTest (from `node-lib/src/runner.rs`). They can also be set from code (i.e not the command line) using the Builder, but to play it safe I'll create an issue to check the network type when creating a new `ChainConfigOptions` instances

Reply:
Because of the way how our command line parsing works, if any of the regtest command line arguments are passed along to other networks, the cli will fail e.g `error: unexpected argument '--chain-pos-netupgrades' found` so I think this should already prevent non-regtest use.

However, https://github.com/mintlayer/mintlayer-core/pull/1092/commits/51789930a744537499582b420bd0994d5436ea03 adds a gate to the regtest options parsing, so at least now in code you must explicitly be passing in the `Command::RegTest`.

Reply:
Looking good!